autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from autogen.agentchat import ConversableAgent
+
+from .graph_store import GraphStore
+
+
+class GraphRagAgent(ConversableAgent, ABC):
+    """
+    A graph rag agent is a conversable agent which could query graph database for answers.
+
+    An implementing agent class would
+    1. create a graph in the underlying database with input documents
+    2. use the retrieve() method to retrieve information.
+    3. use the retrieved information to generate and send back messages.
+    """
+
+    @abstractmethod
+    def _init_db(self, input_doc: List | None = None) -> GraphStore:
+        """
+        This method initializes graph database with the input documents or records.
+        Usually, it takes the following steps,
+        1. connecting to a graph database.
+        2. extract graph nodes, edges based on input data, graph schema and etc.
+        3. build indexes etc.
+
+        return: GraphStore
+        """
+        pass
+
+    @abstractmethod
+    def retrieve(self, question: str, **kwargs):
```

Comment:
Could we reuse as many as API designs in https://github.com/microsoft/autogen/blob/main/autogen/agentchat/contrib/vectordb/base.py ?

This can be `retrieve_docs`.

Reply:
I notice retrieve_docs does not have the same signature as we need here. Create the new apis here and we need to refactor retrieve_docs to make it more generic

Reply:
Same as retrieve